src/Microsoft.DotNet.Build.Tasks/Orchestration/WriteBuildOutputProps.cs

```diff
@@ -0,0 +1,87 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.Build.Framework;
+using NuGet.Packaging.Core;
+using NuGet.Versioning;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Text.RegularExpressions;
+
+namespace Microsoft.DotNet.Build.Tasks.Orchestration
+{
+    public class WriteBuildOutputProps : BuildTask
+    {
+        private const string NuGetPackageInfoId = "PackageId";
+        private const string NuGetPackageInfoVersion = "PackageVersion";
+
+        [Required]
+        public ITaskItem[] NuGetPackageInfos { get; set; }
+
+        [Required]
+        public string OutputPath { get; set; }
+
+        /// <summary>
+        /// Prevents overriding stable packages. Adds a condition to each package version property:
+        /// only set the property when the property already exists and specifies a prerelease.
+        /// </summary>
+        public bool OnlyUpdatePrereleaseVersions { get; set; }
+
+        public override bool Execute()
+        {
+            PackageIdentity[] latestPackages = NuGetPackageInfos
+                .Select(item =>
+                {
+                    var id = item.GetMetadata(NuGetPackageInfoId);
+                    var version = item.GetMetadata(NuGetPackageInfoVersion);
+
+                    if (string.IsNullOrEmpty(id) || string.IsNullOrEmpty(version))
+                    {
+                        return null;
+                    }
+
+                    return new PackageIdentity(id, NuGetVersion.Parse(version));
+                })
+                .Where(identity => identity != null)
+                .GroupBy(identity => identity.Id)
+                .Select(g => g.OrderBy(id => id.Version).Last())
+                .OrderBy(id => id.Id)
+                .ToArray();
+
+            Directory.CreateDirectory(Path.GetDirectoryName(OutputPath));
+
+            var invalidElementNameCharRegex = new Regex(@"(^|[^A-Za-z0-9])(?<FirstPartChar>.)");
```

Comment:
NIt: I'd a quick comment on what does this match

Reply:
Good idea, I just copied this from https://github.com/dotnet/source-build/blob/dev/release/2.0/tools-local/tasks/Microsoft.DotNet.SourceBuild.Tasks/WriteBuildOutputProps.cs but now's a good time to describe it.